homeassistant/components/feedreader/__init__.py

```diff
@@ -46,107 +49,115 @@
-        track_time_interval(
-            hass,
-            lambda now: self._update(),
-            self._scan_interval,
-            cancel_on_shutdown=True,
-        )
-
-    @property
-    def last_update_successful(self) -> bool:
-        """Return True if the last feed update was successful."""
-        return self._last_update_successful
+    async def _async_update(self, _: datetime | Event) -> None:
+        """Update the feed and publish new entries to the event bus."""
+        last_entry_timestamp = await self._hass.async_add_executor_job(self._update)
+        if last_entry_timestamp:
+            self._storage.async_put_timestamp(self._feed_id, last_entry_timestamp)
 
-    def _update(self) -> None:
+    def _update(self) -> struct_time | None:
         """Update the feed and publish new entries to the event bus."""
-        _LOGGER.info("Fetching new data from feed %s", self._url)
+        _LOGGER.debug("Fetching new data from feed %s", self._url)
         self._feed: feedparser.FeedParserDict = feedparser.parse(  # type: ignore[no-redef]
             self._url,
             etag=None if not self._feed else self._feed.get("etag"),
             modified=None if not self._feed else self._feed.get("modified"),
         )
         if not self._feed:
             _LOGGER.error("Error fetching feed data from %s", self._url)
-            self._last_update_successful = False
-        else:
-            # The 'bozo' flag really only indicates that there was an issue
-            # during the initial parsing of the XML, but it doesn't indicate
-            # whether this is an unrecoverable error. In this case the
-            # feedparser lib is trying a less strict parsing approach.
-            # If an error is detected here, log warning message but continue
-            # processing the feed entries if present.
-            if self._feed.bozo != 0:
-                _LOGGER.warning(
-                    "Possible issue parsing feed %s: %s",
-                    self._url,
-                    self._feed.bozo_exception,
-                )
-            # Using etag and modified, if there's no new data available,
-            # the entries list will be empty
-            if self._feed.entries:
-                _LOGGER.debug(
-                    "%s entri(es) available in feed %s",
-                    len(self._feed.entries),
-                    self._url,
-                )
-                self._filter_entries()
-                self._publish_new_entries()
-                if self._has_published_parsed or self._has_updated_parsed:
-                    self._storage.put_timestamp(
-                        self._feed_id, cast(struct_time, self._last_entry_timestamp)
-                    )
-            else:
-                self._log_no_entries()
-            self._last_update_successful = True
-        _LOGGER.info("Fetch from feed %s completed", self._url)
+            return None
```

Comment:
This part of the diff is preferably read with whitespace changes hidden.

Reply:
Uh yeah that helped 👍